test/unit/math/rev/mat/prob/normal_id_glm_lpdf_test.cpp

```diff
@@ -9,8 +9,8 @@ using stan::math::var;
 //  We check that the values of the new regression match those of one built
 //  from existing primitives.
 TEST(ProbDistributionsNormalIdGLM, glm_matches_normal_id_doubles) {
-  Matrix<int, Dynamic, 1> n(3, 1);
-  n << 51, 32, 12;
+  Matrix<int, Dynamic, 1> y(3, 1);
```

Comment:
Quickly looking through the normal_id_glm_lpdf_tests, I think the value/gradient tests look good. The things that are missing are:

1. Tests to make sure the right errors are being thrown
2. Tests to instantiate all the different interface types

X can only be an Eigen::Matrix, right? So that it can be a double or var.

Other than that, cause they all go through scalar_seq_view, each of the other variables can be `double`, `std::vector<double>`, `Eigen::VectorXd`, or `Eigen::RowVectorXd` (or all the same types with vars instead of doubles).

They don't need to be combinatorically tested, and the values don't need to be tested on the output.

So just do something like:
```
double d = 1.0;
std::vector<double> vd = {{ 1.0, 2.0 }};
Eigen::VectorXd ev(2);
Eigen::RowVectorXd rv(2);
Eigen::MatrixXd m1(1, 1);
m1 << 1.0;
Eigen::MatrixXd m(2, 2);
ev << 1.0, 2.0;
rv << 1.0, 2.0;
m << 1.0, 2.0, 3.0, 4.0;

auto a2 = normal_id_glm_lpdf(d, m1, d, d, d);
auto a3 = normal_id_glm_lpdf(vd, m, vd, vd, vd);
auto a4 = normal_id_glm_lpdf(ev, m, ev, ev, ev);
auto a5 = normal_id_glm_lpdf(rv, m, rv, rv, rv);

// Repeat the same thing for vars
```

It's super tedious. But just shovel it out.